src/test/rgw/test_d4n_filter.cc

```diff
@@ -0,0 +1,2810 @@
+    put_object(testName, yield);
+
+    RGWEnv rgw_env;
+    req_info info(env->cct.get(), &rgw_env);
+    rgw_zone_id zone;
+    rgw_placement_rule placement;
+    ceph::real_time mtime;
+    buffer::list bl;
+    bl.append("dest_object_version");
+    rgw::sal::Attrs attrs{{RGW_ATTR_ID_TAG, bl}};
+    std::string tag;
+
+    std::string destName = "dest_object";
+    std::unique_ptr<rgw::sal::Object> destObj = testBucket->get_object(rgw_obj_key(destName));
+    EXPECT_NE(destObj.get(), nullptr);
+
+    int ret = obj->copy_object(acl_owner,
+	     std::get<rgw_user>(owner),
+	     &info,
+	     zone,
+	     destObj.get(),
+	     testBucket.get(),
+	     testBucket.get(),
+	     placement,
+	     &mtime,
+	     &mtime,
+	     nullptr,
+	     nullptr,
+	     false,
+	     nullptr,
+	     nullptr,
+	     rgw::sal::ATTRSMOD_MERGE,
+	     false,
+	     attrs, 
+	     RGWObjCategory::Main,
+	     0,
+	     boost::none,
+	     nullptr,
+	     &tag, 
+	     &tag,
+	     nullptr,
+	     nullptr,
+	     env->dpp,
+	     optional_yield({yield}));
+    EXPECT_EQ(ret, 0);
+
+    boost::system::error_code ec;
+    request req;
+    req.push("EXISTS", TEST_BUCKET + testName + "_" + destName);
+    req.push("EXISTS", TEST_BUCKET + testName + "_" + destName + "_0_0");
+    req.push("EXISTS", TEST_BUCKET + testName + "__:null_" + destName + "_0_0");
+    req.push("EXISTS", TEST_BUCKET + testName + "_" + destName + "_0_9");
+    req.push("ZRANGE", TEST_BUCKET + testName + "_" + destName, "0", "10000000000");
+    req.push("HGETALL", TEST_BUCKET + testName + "_" + destName + "_0_0");
+    req.push("HGETALL", TEST_BUCKET + testName + "__:null_" + destName + "_0_0");
+    req.push("HGETALL", TEST_BUCKET + testName + "_" + destName + "_0_9");
+    req.push("HGET", TEST_BUCKET + testName + "_" + destName + "_0_0", "version");
+    req.push("FLUSHALL");
+
+    response< int, int, int, int, 
+              std::vector<std::string>,
+              std::map<std::string, std::string>,
+              std::map<std::string, std::string>,
+              std::map<std::string, std::string>,
+	      std::string, boost::redis::ignore_t > resp;
+
+    conn->async_exec(req, resp, yield[ec]);
+
+    ASSERT_EQ((bool)ec, false);
+    EXPECT_EQ((int)std::get<0>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<1>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<2>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<3>(resp).value(), 1);
+    EXPECT_EQ(std::get<4>(resp).value()[0], "null");
+    EXPECT_EQ(std::get<5>(resp).value().size(), 10);
+    EXPECT_EQ(std::get<6>(resp).value().size(), 10);
+    EXPECT_EQ(std::get<7>(resp).value().size(), 10);
+
+    std::string version = std::get<8>(resp).value();
+    std::error_code err;
+    std::string testData; 
+    std::ifstream testFile; 
+
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + destName + "/" + version, err), true); // TODO: version here isn't dest version, is this expected? 
```

Comment:
Yes, for write-back cache, the version is calculated internally and is not based on ID tag since the object does not exist in the backend store and ID tag is available only when the object is in backend store and is used in the read cache. Hence version here will not be dest_version set to ID tag.